From 38e4a8236e1fc21b53d8a04f290551796ba4ba10 Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <systemd@esmil.dk>
Date: Thu, 18 Sep 2014 15:24:53 +0200
Subject: [PATCH 14/17] dhcp-internal.h: add fallback IPTOS_CLASS_CS6
 definition

---
 src/libsystemd-network/dhcp-internal.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libsystemd-network/dhcp-internal.h b/src/libsystemd-network/dhcp-internal.h
index 7c60ef1..9bfc37e 100644
--- a/src/libsystemd-network/dhcp-internal.h
+++ b/src/libsystemd-network/dhcp-internal.h
@@ -32,6 +32,10 @@
 #include "sd-dhcp-client.h"
 #include "dhcp-protocol.h"
 
+#ifndef IPTOS_CLASS_CS6
+#define IPTOS_CLASS_CS6 0xc0
+#endif
+
 int dhcp_network_bind_raw_socket(int index, union sockaddr_union *link,
                                  uint32_t xid, const uint8_t *mac_addr,
                                  size_t mac_addr_len, uint16_t arp_type);
-- 
2.1.4

